modules/vault/src/test/java/org/apache/ignite/internal/vault/VaultServiceTest.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.vault;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+import org.apache.ignite.internal.util.Cursor;
+import org.apache.ignite.lang.ByteArray;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+import static org.apache.ignite.internal.vault.CompletableFutureMatcher.await;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+
+/**
+ * Base class for testing {@link VaultService} implementations.
+ */
+public abstract class VaultServiceTest {
+    /** */
+    private static final int TIMEOUT_SECONDS = 1;
+
+    /** Vault. */
+    private VaultService vaultService;
+
+    /** */
+    @BeforeEach
+    public void setUp(TestInfo testInfo) throws IOException {
+        vaultService = getVaultService();
+    }
+
+    /** */
+    @AfterEach
+    public void tearDown() throws Exception {
+        vaultService.close();
+    }
+
+    /**
+     * Returns the vault service that will be tested.
+     */
+    protected abstract VaultService getVaultService();
+
+    /**
+     * Tests regular behaviour of the {@link VaultService#put} method.
+     */
+    @Test
+    public void testPut() throws Exception {
+        ByteArray key = getKey(1);
+
+        assertThat(vaultService.get(key), await(is(new VaultEntry(key, null))));
```

Comment:
This is a minor complaint, but "await is" doesn't resemble a sentence. Maybe we should use something like "willBe(equalTo(...))"?